WRITING_PLUGINS.md

```diff
@@ -2,6 +2,41 @@
 
 Farscape Plugins can be used to add drop-in functionality to Farscape. 
 
+## Farscape Plugin Hash
+
+A Farscape plugin is described with a hash with a set of well defined keys
+
+### name
+This is a symbol representing the name of the plugin
+
+### type
+The type of plugin, this is useful for manipulating multiple plguins simultaneously
+
+### middleware
+Middleware objects as per [Faraday plugins](https://github.com/lostisland/faraday#writing-middleware)
+
+### extensions
+A list of classes which will extend the default functionality of Farscape clases.
+
+### extends
+A list of classes which will be extended by :extensions
+Possible extends values are: :Agent, :HttpClient, :SafeRepresentorAgent, :RepresentorAgent, and :TransitionAgent
+
+### default_state
+When registered, is this :enabled or :disabled by default
+
+### Example
+```ruby
+{
+  name: :PluginName
+  type: :ExampleType
+  middleware: PluginMiddlewareClass
+  extension: [ExtentionClass]
```

Comment:
So instead of applying every extension to every extended class, why don't we allow this to be specified? something like
```ruby
{ ...
  extenstions: { :Agent => [ExtenstionClass],
                         :RepresentorAgent => [ExtensionClass, Walrus]
                      }
... }